File: 알고리즘/정렬/bubble_sort.py
import random

def bubble_sort(data):
    target: list = data

    for i in range(len(target) - 1):
        is_swap: bool = False
        for j in range((len(target) - i) - 1):
            if target[j] > target[j + 1]:
                is_swap = True
                target[j], target[j + 1] = target[j + 1], target[j]
            
        if is_swap == False:
                break
    return target

data_list: list = random.sample(range(100), 50)

print(bubble_sort(data_list))

File: 알고리즘/정렬/bubble_sort.js
// bubble sort

function bubbleSort() {
  const array = [80, 30, 90, 40, 50, 70, 60, 20, 10, 100, 50];

  for (let i = 0; i < array.length - 1; i++) {
    let isSwap = false;
    for (let j = 0; j < array.length - 1 - i; j++) {
      if (array[j] > array[j + 1]) {
        isSwap = true;
        [array[j], array[j + 1]] = [array[j + 1], array[j]];
      }
    }
    if (!isSwap) break;
  }

  return array;
}

console.log(bubbleSort());
